### ISSUE TITLE
Forbid methods with no arguments

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#367]
# Rule request

## Thesis

Currently we allow the following code:

```python
class Test(object):
    def area():
        return 1
```

But, we clearly should not do that.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/method_has_no_argument.html

How to handle this situation? We need to check that all methods have at least one argument.
`@staticmethod` can have 0-arguments for sanity. We should check for it directly.

## Reasoning

This is clearly a mistake.

**Comment by sobolevn** [comment]
On the other hand, why should we make a difference for `@staticmethod`s? First of all, we do not allow it. Secondly, it is kind of strange to not accept any arguments at all even for a `@staticmethod`.

So, there would be no exception for this case.
